### DISCUSSION TITLE
Surface-fitting functionality

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#670] [Q&A] [answered]
Hi Marco, 
Just a quick question: does vedo have the functionality to fit a surface mesh to a series of points? I feel like the answer to this is yes...haha. do you have any examples you could point me too. Also are there any particular techniques you use to accomplish this? 

Cheers
j

**Comment by marcomusy** [comment, marked as answer]
Yes :)
You can use `warp`:
```python
from vedo import Grid, Points, Arrows, show
import numpy as np
np.random.seed(2)

surf = Grid([0,0,0], res=[25,25])
idxs = np.random.randint(0, surf.npoints, 10)  # pick 10 indices
pts = surf.points(idxs)

ptsource, pttarget = [], []
for pt in pts:
    pt1 = pt + [0, 0, np.random.randn(1)[0] * 0.1]
    pt2 = surf.closest_point(pt1)
    ptsource.append(pt2)
    pttarget.append(pt1)

warped = surf.warp(ptsource, pttarget, mode='2d')
warped.color("b4").lc('light blue').wireframe(False).lw(1)

apts = Points(pttarget, r=15, c="r")
arrs = Arrows(ptsource, pttarget, c='k')

show(warped, apts, arrs, axes=1, viewup="z").close()
```

![Screenshot from 2022-07-19 11-43-15](https://user-images.githubusercontent.com/32848391/179720912-e533833e-dc89-4837-aaa2-8c652e1d593a.png)


Note that this is more of a "splining" than a fitting, for a fitting you must need to provide a model.
